### ISSUE TITLE
Logging tests race condition?

### ISSUE THREAD

**Issue by hugo-vrijswijk** [stryker-mutator/stryker4s#44]
The logging tests are failing randomly sometimes on the build-server. I think there's a race condition/timing issue in the current code. SBT runs tests in parallel, and I think resetting the test log causes the issue.
Here's what I think happens:

TEST A: Starts test
TEST B: Starts test
TEST A: Logs a statement
TEST B: Logs a statement
TEST A: Assert for logged statement
TEST A: Finished, **resets TestAppender**. Test succeeds!
TEST B: Assert for logged statement. Not found in the TestAppender. Test fails!

**Comment by legopiraat** [comment]
Yes this is the case since the events are stored in an object.

**Comment by legopiraat** [comment]
I've had a random idea on how we could potentially solve this issue.
We could extend the log matchers that receive an implicit class tag or normal string so we know which class is under test. Then we could loop the events and see if that specific class logged something. 

This will gain us the advantage that we won't leak log events to other tests. And we would be able to clean up the log events based on the class under test so we won't leak log statements over multiple test cases. Which would be a problem if we won't do any cleanup.

I'll see if i can get it working later this weekend.
